diff -Naur work.orig/vmmon-only/linux/driver.c work/vmmon-only/linux/driver.c
--- work.orig/vmmon-only/linux/driver.c	2015-02-12 14:10:10.469007755 +0100
+++ work/vmmon-only/linux/driver.c	2015-02-12 14:11:41.815631517 +0100
@@ -1328,7 +1328,7 @@
  *-----------------------------------------------------------------------------
  */
 
-__attribute__((always_inline)) static Bool
+INLINE_SINGLE_CALLER static Bool
 LinuxDriverSyncReadTSCs(uint64 *delta) // OUT: TSC max - TSC min
 {
    TSCDelta tscDelta;
