diff -urpN alsa-lib-1.0.28.orig/src/control/namehint.c alsa-lib-1.0.28/src/control/namehint.c
--- alsa-lib-1.0.28.orig/src/control/namehint.c	2014-06-17 05:34:19.000000000 -0700
+++ alsa-lib-1.0.28/src/control/namehint.c	2014-10-01 04:45:10.616849789 -0700
@@ -553,6 +553,10 @@ int snd_device_name_hint(int card, const
 	snd_config_iterator_t i, next;
 	int err;
 
+	if (card == -1) {
+		fprintf(stderr, "Dropping snd_device_name_hint because card == -1\n");
+		return -EINVAL;
+	}
 	if (hints == NULL)
 		return -EINVAL;
 	err = snd_config_update_r(&local_config, &local_config_update, NULL);
